lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,46 @@
+module MigrationSyncHelper
+  def self.migrations_column_present?
+    ActiveRecord::Base.connection.columns("miq_regions").map(&:name).include?("migrations_ran")
+  end
+
+  def self.wait_for_remote_region_migrations(region, version, wait_time = 1)
+    return unless MigrationSyncHelper.migrations_column_present?
+    until provider_region.migrations_ran.include?(version)
+      s.disable
+      s.enable
+      sleep wait_time
+      provider_region.reload
+    end
+  end
+
+  def self.update_local_migrations_ran(version, direction)
+    return unless MigrationSyncHelper.migrations_column_present?
+    region = MiqRegion.my_region
+    if region
+      new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+      new_migrations << version.to_s if direction == :up
+
+      ActiveRecord::Base.connection.exec_query(<<~SQL)
+        UPDATE #{MiqRegion.table_name}
+        SET migrations_ran = '#{PG::TextEncoder::Array.new.encode(new_migrations)}'
+        WHERE id = #{region.id}
+      SQL
+    end
+  end
+end
+
+module ArPglogicalMigration
+  def migrate(direction)
+    PglogicalSubscription.all.each do |s|
+      MigrationSyncHelper.wait_for_remote_region_migration(MiqRegion.find(s.provider_region), version)
+    end
+
+    ret = super
```

Comment:
`super.tap`?

Reply:
Eh, kind of would rather this version because we wouldn't be using the return value of `super` in the `tap` block.
I think the way I have it is more explicit about returning the previous value where using `tap` feels more like taking advantage of a side-effect.